integration-test/src/test/scala/com/cloudera/livy/test/BatchIT.scala

```diff
@@ -170,89 +160,17 @@ class BatchIT extends BaseIntegrationTestSuite with BeforeAndAfterAll {
     hdfsPath.toUri().getPath()
   }
 
-  private def runScript(script: String, args: List[String] = Nil): SessionInfo = {
-    val request = new CreateBatchRequest()
-    request.file = script
-    request.args = args
-    waitAndDeleteBatch(startBatch(request).id)
-  }
-
-  private def runSpark(
-      klass: Class[_],
-      args: List[String] = Nil,
-      waitForExit: Boolean = true): SessionInfo = {
-    val request = new CreateBatchRequest()
-    request.file = testLibPath
-    request.className = Some(klass.getName)
-    request.args = args
-
-    val batchId = startBatch(request)
-    if (waitForExit) waitAndDeleteBatch(batchId.id) else batchId
-  }
-
-  private def getBatchSessionInfo(batchId: Int): SessionInfo = {
-    val r = httpClient.prepareGet(s"$livyEndpoint/batches/$batchId")
-      .execute()
-      .get()
-    assert(r.getStatusCode() === SC_OK)
-    mapper.readValue(r.getResponseBodyAsStream(), classOf[SessionInfo])
-  }
-
-  private def deleteBatch(batchId: Int): Unit = {
-    val r = httpClient.prepareDelete(s"$livyEndpoint/batches/$batchId")
-      .execute()
-      .get()
-    assert(r.getStatusCode() === SC_OK)
-  }
-
-  private def startBatch(request: CreateBatchRequest): SessionInfo = {
-    request.conf = Map("spark.yarn.maxAppAttempts" -> "1")
-
-    val response = httpClient.preparePost(s"$livyEndpoint/batches")
-      .setBody(mapper.writeValueAsString(request))
-      .execute()
-      .get()
-
-    withClue(response) {
-      assert(response.getStatusCode === SC_CREATED)
-    }
-
-    mapper.readValue(response.getResponseBodyAsStream(), classOf[SessionInfo])
-  }
-
-  private def waitUntilRunning(batchId: Int): ApplicationId = {
-    eventually {
-      val batch = getBatchSessionInfo(batchId)
-
-      // If batch state transits to any error state, fail test immediately.
-      if (batch.state == SessionState.Error().toString() ||
-        batch.state == SessionState.Dead().toString()) {
-        fail(s"Session shouldn't be in a terminal state: ${batch.state}")
-      }
-
-      assert(batch.state === SessionState.Running().toString())
-
-      ConverterUtils.toApplicationId(batch.appId)
-    }
+  private def withScript[R]
+    (scriptPath: String, args: List[String], sparkConf: Map[String, String] = Map.empty)
+    (f: (LivyRestClient#BatchSession) => R): R = {
```

Comment:
Is `LivyRestClient#BatchSession` the same as `LivyRestClient.BatchSession`? Never seen that syntax. If it's the same I prefer the latter.